server/api.go

```diff
@@ -442,6 +442,34 @@ func (a *API) getMSTeamsTeamChannels(w http.ResponseWriter, r *http.Request) {
 	a.writeJSON(w, http.StatusOK, paginatedChannels)
 }
 
+func (a *API) linkChannels(w http.ResponseWriter, r *http.Request) {
+	userID := r.Header.Get(HeaderMattermostUserID)
+
+	var body *storemodels.ChannelLink
+	if err := json.NewDecoder(r.Body).Decode(&body); err != nil {
+		a.p.API.LogError("Error occurred while decoding link channels payload.", "Error", err.Error())
+		http.Error(w, "Error occurred while decoding link channels payload.", http.StatusInternalServerError)
+		return
+	}
+
+	if err := body.IsChannelLinkPayloadValid(); err != nil {
+		a.p.API.LogError("Invalid channel link payload.", "Error", err.Error())
+		http.Error(w, "Invalid channel link payload.", http.StatusBadRequest)
+		return
+	}
+
+	if errMsg, statusCode := a.p.LinkChannels(userID, body.MattermostTeamID, body.MattermostChannelID, body.MSTeamsTeamID, body.MSTeamsChannelID); errMsg != "" {
```

Comment:
make sure we are logging the error msg in Link Channels if not we can add the logs here

Reply:
yes we are!